src/catch2/matchers/catch_matchers_generic.hpp

```diff
@@ -0,0 +1,117 @@
+
+//              Copyright Catch2 Authors
+// Distributed under the Boost Software License, Version 1.0.
+//   (See accompanying file LICENSE_1_0.txt or copy at
+//        https://www.boost.org/LICENSE_1_0.txt)
+
+// SPDX-License-Identifier: BSL-1.0
+#ifndef CATCH_MATCHERS_GENERIC_HPP_INCLUDED
+#define CATCH_MATCHERS_GENERIC_HPP_INCLUDED
+
+#include <catch2/matchers/catch_matchers_templated.hpp>
+
+#include <algorithm>
+#include <functional>
+#include <utility>
+
+namespace Catch {
+    namespace Matchers {
+        // Matcher for checking that all elements in range matches a given matcher.
+        template <typename Matcher>
+        class AllMatchMatcher final : public MatcherGenericBase {
+            Matcher m_matcher;
+        public:
+            AllMatchMatcher(Matcher matcher):
+                m_matcher(std::move(matcher))
+            {}
+
+            std::string describe() const override {
+                return "all match " + m_matcher.describe();
+            }
+
+            template <typename RangeLike>
+            bool match(RangeLike&& rng) const {
+                using std::begin; using std::endl;
```

Comment:
You shouldn't need these, range-for already does the requisite ADL lookup

Reply:
You are right!! Haha thanks